crates/matrix-sdk-base/src/store/mod.rs

```diff
@@ -507,8 +507,11 @@ impl StoreConfig {
             #[cfg(feature = "e2e-encryption")]
             crypto_store: matrix_sdk_crypto::store::MemoryStore::new().into_crypto_store(),
             state_store: Arc::new(MemoryStore::new()),
-            event_cache_store: crate::event_cache_store::MemoryStore::new()
-                .into_event_cache_store(),
+            event_cache_store: event_cache_store::EventCacheStoreLock::new(
+                event_cache_store::MemoryStore::new(),
+                "default-key".to_owned(),
+                "matrix-sdk-base".to_owned(),
```

Comment:
The values provided here are scary, and should be controllable by the callers, otherwise it's not possible to distinguish a process from another. The key could be a constant used everywhere in the codebase (maybe buried into a single place, and then passed around — or buried a single time when creating an `EventCacheStoreLock` from an event cache store impl), but the holder's value has to be provided by the embedder.

Reply:
I would like to address that as a follow-up PR: I want to introduce a "process holder name" inside the client that can be used by all cross-process lock, otherwise things start to be really clumsy. What do you think?

Reply:
@bnjbvr and I have agreed to remove these values with a configuration in the `Client`. I'm already working on this as a follow-up PR.

Reply:
Here it is, https://github.com/matrix-org/matrix-rust-sdk/pull/4224